src/lib/WalletInfoCollector.ts

```diff
@@ -92,20 +94,39 @@ export default class WalletInfoCollector {
             derivedAccountsPromise = WalletInfoCollector._deriveAccounts(startIndex,
                 ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, walletType, walletId);
 
-            // find accounts with a balance > 0 and/or transaction receipts
-            foundAccounts = [];
+            // find accounts with a balance > 0
+            const foundAddresses: string[] = [];
             const balances = await WalletInfoCollector._getBalances(derivedAccounts!);
             for (const account of derivedAccounts!) {
                 const balance = balances.get(account.address);
                 if (balance !== undefined && balance !== 0) {
-                    foundAccounts.push(account);
-                    continue;
-                }
-                const receipts = await NetworkClient.Instance.requestTransactionReceipts(account.address);
-                if (receipts.length > 0) {
-                    foundAccounts.push(account);
+                    foundAddresses.push(account.address);
                 }
             }
+
+            // for accounts with balance 0 check if there are transactions
+            const addressesToCheck = derivedAccounts.map((account) => account.address)
```

Comment:
That's unreasonable. There is no measurable difference between accessing an object property and an array directly. Since there is no computation involved in getting the address here, I think adding yet another variable is unnecessary.

Reply:
Of course there is no performance difference, it's a question of code style.